packages/vite/src/node/plugins/css.ts

```diff
@@ -1406,10 +1406,12 @@ async function compileCSS(
         // https://github.com/postcss/postcss/blob/main/docs/guidelines/plugin.md#3-dependencies
         const { dir, glob: globPattern = '**' } = message
         const pattern =
-          glob.escapePath(normalizePath(path.resolve(path.dirname(id), dir))) +
+          escapePath(normalizePath(path.resolve(path.dirname(id), dir))) +
           `/` +
           globPattern
-        const files = glob.sync(pattern, {
+        const files = globSync(pattern, {
+          absolute: true,
```

Comment:
Due to the way we construct the pattern above, we expect absolute paths here, but `tinyglobby` (through the underlying `fdir`) returns matches relative to CWD even with an absolute pattern, so we force the output paths to be absolute through this option.

Reply:
FYI you might want to set the `cwd` to the dir before `globPattern`, otherwise optimizations might be skipped and things could be heavily slowed down 😅

if you do that you can also avoid having to use an absolute path as a pattern

Reply:
@sapphi-red Don't we need absolute matches here for the deps to work correctly?

Reply:
We need the returned `files` to be absolute. I thought we can utilize the `cwd` option like:
```ts
const files = globSync(globPattern, {
  cwd: path.resolve(path.dirname(id), dir),
  absolute: true,
  expandDirectories: false,
  ignore: ['**/node_modules/**'],
})
```
as @SuperchupuDev suggested.

Reply:
Ah, I see, my bad! Pushed the suggested change.